core/base/document.php

```diff
@@ -1284,4 +1284,23 @@ protected function get_post_statuses() {
 	protected function get_have_a_look_url() {
 		return $this->get_permalink();
 	}
+
+	public function handle_revisions_changed() {
+		add_filter( 'wp_save_post_revision_post_has_changed', function ( $post_has_changed, $last_revision, $post ) {
```

Comment:
Convert the function to a method andn remove the filter after the document is saved.